### PR TITLE
v3.0

### PR DESCRIPTION
**Pull request by naokazuterada**
Hi, @jonasbn !

I finally got ready for MarkdownTOC v3.0.
[As I mentioned before](https://github.com/naokazuterada/MarkdownTOC/pull/114), it contains some breaking changes. I summarized them in the [Upgrade Guide](https://github.com/naokazuterada/MarkdownTOC/blob/v3/messages/3.0.0.md#upgrade-guide) on message file.

Any other changes are also summarized on #110 .

I am very grateful if you give me your opinion on especially English sentences and how to organize documents. Please let me know if you feel strange or confusing points. I think that you are busy though, I hope that you could take a time.

### CODE REVIEW

**Comment by naokazuterada** [comment]
Sorry, I found there were places to update, so I fixed it.

**Comment by jonasbn** [comment]
Hi @naokazuterada 

I will have a look ASAP

File: messages/2.8.0.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# MarkdownTOC - 2.8.0
+
+## Changes
+
+- Allow regular expressions in 'id_replacements' Ref: #113
+
+## Notice
+
+Setting format has been changed.
+You have to modify `id_replacements` section if you customised `MarkdownTOC.sublime-settings` by yourself.
```
Comment (jonasbn) [inline messages/2.8.0.md]: Suggestion: remove "by" , so it reads "`MarkdownTOC.sublime-settings` yourself."

File: messages/3.0.0.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
+# MarkdownTOC - 3.0.0
+
+---
+
+**NOTE**
+This upgrade contains some **breaking changes**.
```
Comment (jonasbn) [inline messages/3.0.0.md]: Suggestion: "This upgrade contains **breaking changes**. The fact is more relevant, than the amount.

File: messages/3.0.0.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
+# MarkdownTOC - 3.0.0
+
+---
+
+**NOTE**
+This upgrade contains some **breaking changes**. 
+Please follow "Upgrade guide from v2 to v3" instructions.
```
Comment (jonasbn) [inline messages/3.0.0.md]: Suggestion: "Please follow "Upgrade guide from v2 to v3" for instructions."

File: messages/3.0.0.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+You can also override it as attribute
+
+```markdown
+<!-- MarkdownTOC levels="1,2,3" -->
+```
+
+Note: 
+
+- Values type is 'conmma separated string'
```
Comment (jonasbn) [inline messages/3.0.0.md]: Minor spelling errors: "conmma" should read: "comma" and "string" should be plural as: "strings"

File: messages/3.0.0.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+- The default value changed to `[1,2,3,4,5,6]`, it means all headings(`h1`...`h6`) appear in TOC. So if you don't like this, please override in your **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`)
```
Comment (jonasbn) [inline messages/3.0.0.md]: Suggestion: Remove "So if you don't like this" and perhaps write: " please override in your **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`) if you have different preferences."

File: messages/3.0.0.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+### 5. `defaults.bracket`(ex `default_bracket`) default value is changed to `"round"`
+
+The default value has been changed. So if you prefer `"square"` style, please override in your **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`)
+
+
+### 6. Allow regular expressions in 'id_replacements', the format has been changed
+
+Setting format has been changed.
+You have to modify `id_replacements` section if you customised **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`) by yourself.
```
Comment (jonasbn) [inline messages/3.0.0.md]: Suggestion: remove "by" , so it reads "MarkdownTOC.sublime-settings yourself."

File: messages/2.8.0.md (hunk 1 above)
Comment (naokazuterada) [reply messages/2.8.0.md]: 👍 thanks

File: messages/3.0.0.md (hunk 3 above)
Comment (naokazuterada) [reply messages/3.0.0.md]: I see.

File: messages/3.0.0.md (hunk 4 above)
Comment (naokazuterada) [reply messages/3.0.0.md]: OK! thanks 👀

File: messages/3.0.0.md (hunk 5 above)
Comment (naokazuterada) [reply messages/3.0.0.md]: Thanks!
How about this correction?

So if you don't like this, please override in your **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`)
↓
Please override in your **User Settings**(`Packages/User/MarkdownTOC.sublime-settings`) if you prefer previous settings (e.g. `[1,2]`)

Comment (jonasbn) [reply messages/3.0.0.md]: Sounds good 👍

Comment (naokazuterada) [reply messages/3.0.0.md]: 💃

File: messages/3.0.0.md (hunk 2 above)
Comment (naokazuterada) [reply messages/3.0.0.md]: I couldn't understand well, sorry...

Which is your suggestion?

1:
This upgrade contains some **breaking changes**.
↓
This upgrade contains **breaking changes**.

or

2:
This upgrade contains some **breaking changes**.
↓
This upgrade contains **breaking changes**. The fact is more relevant, than the amount.

Comment (jonasbn) [reply messages/3.0.0.md]: Loose the "some", I would go for: "This upgrade contains breaking changes."

Comment (naokazuterada) [reply messages/3.0.0.md]: OK, I got it!

**Comment by naokazuterada** [comment]
@jonasbn Thank you so much for your polite review.
I've done correcting, and will go for release soon. 👍

**Pull request merged**